### ISSUE TITLE
COG: mismatch of reported ghost area size

### ISSUE THREAD

**Issue by constantinius** [OSGeo/gdal#2333]
## Expected behavior and actual behavior.

The ghost area of a newly generated COG reports a wrong byte size:

```
GDAL_STRUCTURAL_METADATA_SIZE=000140 bytes
LAYOUT=IFDS_BEFORE_DATA
BLOCK_ORDER=ROW_MAJOR
BLOCK_LEADER=SIZE_AS_UINT4
BLOCK_TRAILER=LAST_4_BYTES_REPEATED
KNOWN_INCOMPATIBLE_EDITION=NO
```

This is actually `181` bytes and not `140`.

## Steps to reproduce the problem.

Using the processed LZW file from geotiff.js downloaded/generated [here](https://github.com/geotiffjs/geotiff.js/blob/master/test/data/setup_data.sh#L10).
`gdal_translate -of COG data/lzw.tiff data/cog.tif`

## Operating system

Docker image osgeo/gdal:latest

## GDAL version and provenance

Using the docker image osgeo/gdal:latest with the internal `gdalinfo --version`:
GDAL 3.1.0dev-20569c002c19721d3eb6f7c276fdc7a43cdfd31d, released 2020/03/20

**Comment by rouault** [comment]
The indicated size is the size after "GDAL_STRUCTURAL_METADATA_SIZE=XXXXXX bytes" itself (which is of fixed size). Maybe the doc should make it clearer

**Comment by constantinius** [comment]
Okay, then the documentation seems to be wrong, as it states:

> The ghost area starts with GDAL_STRUCTURAL_METADATA_SIZE=XXXXXX bytesn where XXXXXX describes the size of this whole section (starting at the beginning of GDAL_STRUCTURAL_METADATA_SIZE).

As I read this, the  `GDAL_STRUCTURAL_METADATA_SIZE=000140 bytes` are part of the number of bytes.

**Comment by rouault** [comment]
I've committed the following doc fix:
```
The ghost area starts with ``GDAL_STRUCTURAL_METADATA_SIZE=XXXXXX bytes\n`` (of
a fixed size of 43 bytes) where XXXXXX is a 6-digit number indicating the remaining
size of the section (that is starting after the linefeed character of this starting
line).
```
As this has been implemented more than 6 months ago and there are files already circulating with this layout, I prefer keeping the current implementation. Another reason is that when you've variable size content in a file, the preceding record generally indicates the remaining size in my experience, not the total size. So here the logic is "read 43 bytes and check if they are starting with GDAL_STRUCTURAL_METADATA_SIZE=, then take atoi() of the number after = sign, and read the number of bytes indicated".
